Casks/b/beeper.rb

```diff
@@ -1,39 +1,29 @@
 cask "beeper" do
   arch arm: "arm64", intel: "x64"
 
-  version "3.110.1,241031sx3gh9lyl"
-  sha256 arm:   "632e61ba8ab9a9648cfcace1cfa89f53ad184398f647b70ff777ee0576a834d5",
-         intel: "2735ca5517b34be992424e2a45dfe448184bf40bd987f45499db09610ac3a7f6"
+  version "4.0.604"
+  sha256 :no_check
 
-  url "https://download.todesktop.com/2003241lzgn20jd/Beeper%20#{version.csv.first}%20-%20Build%20#{version.csv.second}-#{arch}-mac.zip",
-      verified: "download.todesktop.com/2003241lzgn20jd/"
+  url "https://api.beeper.com/desktop/download/macos/#{arch}/stable/com.automattic.beeper.desktop"
   name "Beeper"
   desc "Universal chat app powered by Matrix"
   homepage "https://www.beeper.com/"
 
   livecheck do
-    url "https://download.todesktop.com/2003241lzgn20jd/latest-mac.yml"
-    regex(/Beeper\sv?(\d+(?:\.\d+)+)(?:\s-\sBuild\s([a-z\d]+?))?-#{arch}-mac\.zip/i)
-    strategy :electron_builder do |yaml, regex|
-      yaml["files"]&.map do |item|
-        match = item["url"]&.match(regex)
-        next if match.blank?
-
-        match[2].present? ? "#{match[1]},#{match[2]}" : match[1]
-      end
-    end
+    regex(/Beeper-(\d+(?:\.\d+)+)-mac\.zip/i)
```

Comment:
```suggestion
    regex "https://api.beeper.com/desktop/download/macos/#{livecheck_arch}/stable/com.automattic.beeper.desktop
```

We shouldn't need to provide a `regex` here as the default for the `HeaderMatch` strategy should detect the version automatically.

Reply:
Yeah, I realized that right before the review. I was basing the changes on the [commit](https://github.com/Homebrew/homebrew-cask/commit/9fa17b1b627bc296fdf659069cd71bfd48f8183e) that switched from an unversioned url to a versioned one and thought the regex was mandatory; should've read the docs instead :) Fixed.